petab_select/candidate_space.py

```diff
@@ -529,6 +602,628 @@ def iterate():
+        # the current method is swap, check the swap move is contained
+        # in a swap set.
+
+        if self.check_critical(model) and (
+            not self.method == Method.LATERAL or self.check_swap(model)
+        ):
+            return_value = self.inner_candidate_space.consider(model)
+
+            # update the attributes
+            self.models = self.inner_candidate_space.models
+            self.distances = self.inner_candidate_space.distances
+            self.exclusions = self.inner_candidate_space.exclusions
+
+            return return_value
+
+        return True
+
+    def _consider_method(self, model) -> bool:
+        """See `CandidateSpace._consider_method`."""
+        return self.inner_candidate_space._consider_method(model)
+
+    def reset_accepted(self) -> None:
+        """Changing the reset_accepted to reset the
+        inner_candidate_space as well."""
+        self.models = []
+        self.distances = []
+        self.inner_candidate_space.reset_accepted()
+
+    def set_predecessor_model(
+        self, predecessor_model: Union[Model, str, None]
+    ):
+        """Setting the predecessor model for the
+        inner_candidate_space as well."""
+        super().set_predecessor_model(predecessor_model=predecessor_model)
+        self.inner_candidate_space.set_predecessor_model(
+            predecessor_model=predecessor_model
+        )
+
+    def set_exclusions(self, exclusions: Union[List[str], None]):
+        """Setting the exclusions for the
+        inner_candidate_space as well."""
+        self.exclusions = exclusions
+        self.inner_candidate_space.exclusions = exclusions
+        if self.exclusions is None:
+            self.exclusions = []
+            self.inner_candidate_space.exclusions = []
+
+    def set_limit(self, limit: TYPE_LIMIT = None):
+        """Setting the limit for the
+        inner_candidate_space as well."""
+        if limit is not None:
+            self.limit.set_limit(limit)
+            self.inner_candidate_space.limit.set_limit(limit)
+
+    def is_plausible(self, model: Model) -> bool:
+        if self.check_critical(model):
+            return self.inner_candidate_space.is_plausible(model)
+        return False
+
+    def check_swap(self, model: Model) -> bool:
+        """Check if parameters that are swapped are contained in the
+        same swap parameter set."""
+        if self.method != Method.LATERAL:
+            return True
+
+        predecessor_estimated_parameters_ids = set(
+            self.predecessor_model.get_estimated_parameter_ids_all()
+        )
+        estimated_parameters_ids = set(model.get_estimated_parameter_ids_all())
+
+        swapped_parameters_ids = estimated_parameters_ids.symmetric_difference(
+            predecessor_estimated_parameters_ids
+        )
+
+        for swap_set in self.swap_parameter_sets:
+            if swapped_parameters_ids.issubset(set(swap_set)):
+                return True
```

Comment:
Is the following supported? i.e., a lateral move that performs two swaps simultaneously, each from a different swap set
- swap parameters `1` and `2`
- swap parameters `3` and `4`
- swap sets: `[{1, 2}, {3, 4}]`

i.e. all parameters do not belong to the same swap set, so this would not `return True`, right?

Reply:
Might be difficult to support, not sure how to handle best

Reply:
So currently no. Because famos currently always uses LateralCandidateSpace with max_number_of_swaps = 1. So this cannot happen for famos. But it can be implemented. 
But yeah, it would be weird how to check the swap. The only information we have is the difference between model and predecessor_model. So we don't know the swap sets, but we know removed and added parameters. 
So I guess we would have to iterate over possible combinations like:

Removed {1, 3} and added {2, 4}. 
Generate all combinations of parameter swaps: 
[[{1, 4},{3, 2}], [{1, 2},{3, 4}]] 
i.e. first combination is that we swapped 4 for 1, and 2 for 3. (wrong one). Second... 
 
And then for each of these combinations iterate through the swap sets of that combination to check whether each parameter swap of a combination is inside a "swap_set of self.swap_parameter_sets".

If you want I can implement this easily. But then we would also need to add another famos option called `max_lateral_number_of_swaps` so that it allows lateral to do more than one swap in a step.